src/game/ObjectMgr.cpp

```diff
@@ -1975,6 +1975,38 @@ PetLevelInfo const* ObjectMgr::GetPetLevelInfo(uint32 creature_id, uint32 level)
     return &itr->second[level - 1];                         // data for level 1 stored in [0] array element, ...
 }
 
+PetFamilyStatMods const* ObjectMgr::GetPetFamilyStatMods(uint32 family) const
+{
+    // Loading levels data
+    //                                                   0      1     2     3
+    QueryResult* result  = WorldDatabase.Query("SELECT family, hmod, amod, dmod FROM pet_familystats");
+
```

Comment:
Here my proposal for that table:
```SQL
 DROP TABLE IF EXISTS `pet_familystats`;
 CREATE TABLE `pet_familystats` (
  `Family` mediumint(8) unsigned NOT NULL DEFAULT '0',
  `HealthModifier` float NOT NULL DEFAULT '1',
  `DamageModifier` float NOT NULL DEFAULT '1',
  `ArmorModifier` float NOT NULL DEFAULT '1',
   PRIMARY KEY (`Family`)
 ) ENGINE=MyISAM DEFAULT CHARSET=utf8 ROW_FORMAT=FIXED COMMENT='Pet System';
```
Nothing very fancy as you can see. :)
I gave a more explicit name for the fields, you may want to adjust the SQL query from core accordingly.

Also, it seems you query the database every time a pet is summoned. The table will probably be very small, so we can think of loading it in core at startup like what was done here: https://github.com/cmangos/mangos-classic/commit/1e9adf8c86d5e9cfbc702fc1ced27e359bfbeca0 for creatures' stats. This would allow: to reduce query to the database and prevent any issue with this (like connection dropped), check the table consistency/existence only once and for good. What do you think?
Poke @cyberium for his opinion.